scripts/run-e2e-test.sh

```diff
@@ -0,0 +1,134 @@
+}
+
+get_pod_log() {
+  kubectl logs --namespace "${namespace}" "Job/${name}" "$@" || true
+}
+
+echo "$0: deleting any resources left over from the previous run..."
+cleanup 2> /dev/null || true
+
+echo "$0: creating resource to run the suite..."
+kubectl create --namespace "${namespace}" --filename - <<EOF
+apiVersion: v1
+kind: List
+items:
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: "${name}"
+      namespace: "${namespace}"
+      labels:
+        name: "${name}"
+  - apiVersion: rbac.authorization.k8s.io/v1beta1
+    kind: ClusterRole
+    metadata:
+      name: "${name}"
+      labels:
+        name: "${name}"
+    rules:
+      - apiGroups: [ '*' ]
+        resources: [ '*' ]
+        verbs: [ '*' ]
+      - nonResourceURLs: [ '*' ]
+        verbs: [ '*' ]
+  - apiVersion: rbac.authorization.k8s.io/v1beta1
+    kind: ClusterRoleBinding
+    metadata:
+      name: "${name}"
+      namespace: "${namespace}"
+      labels:
+        name: "${name}"
+    roleRef:
+      kind: ClusterRole
+      name: cluster-admin
+      apiGroup: rbac.authorization.k8s.io
+    subjects:
+      - kind: ServiceAccount
+        name: "${name}"
+        namespace: "${namespace}"
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: "${name}"
+      namespace: "${namespace}"
+      labels:
+        name: "${name}"
+    spec:
+      backoffLimit: 0
+      template:
+        metadata:
+          labels:
+            name: "${name}"
+        spec:
+          serviceAccount: "${name}"
+          tolerations:
+            - effect: NoSchedule
+              operator: Exists
+          restartPolicy: Never
+          containers:
+            - name: "${name}"
+              image: "linuxkit/${name}:260c4467ab0a752b5369e73d4f2da13eb2da7bbf"
+              imagePullPolicy: IfNotPresent
+              env:
+              - name: E2E_CLOUD_PROVIDER
+                value: "${E2E_CLOUD_PROVIDER}"
+              - name: E2E_PARALLEL
+                value: "${E2E_PARALLEL}"
+              - name: E2E_FLAKE_ATTEMPTS
+                value: "${E2E_FLAKE_ATTEMPTS}"
+              - name: E2E_FOCUS
+                value: "${E2E_FOCUS}"
+              - name: E2E_SKIP
+                value: "${E2E_SKIP}"
+EOF
+
+echo "$0: waiting until 1 pod is running..."
+until [ "$(get_running_pod | wc -l)" -eq 1 ] ; do sleep 0.5 ; done
+get_pod_log --follow
+echo "$0: kubectl logs terminated, waiting until no pods are running..."
+## sometimes kubectl logs terminates abruptly, and even in theory
+## we cannot trust it to indicated overall status, so we wait, and
+## dump the logs once again after the pod has exited
+until [ "$(get_running_pod | wc -l)" -eq 0 ] ; do sleep 0.5 ; done
+get_pod_log | tee e2e.log
+echo "$0: log save in ${PWD}/e2e.log, cleaning up the resources..."
+cleanup 2> /dev/null || true
+grep -q '^Test Suite Passed$' e2e.log && exit 0
+grep -q '^Test Suite Failed$' e2e.log && exit 1
```

Comment:
If the log contains neither the pass nor fail messages that is a failure IMHO (perhaps the file is empty because pod logging is b0rked), I think you can just drop the `grep` from this last line and `exit 1` unconditionally.